### DISCUSSION TITLE
Vitest faster testing approach?

### DISCUSSION THREAD

**Discussion by karthxk07** [vitest-dev/vitest#5669] [Q&A]
I am using vitest for testing on my project. But it takes up a lot of time(~20 min) for every PR made.

I was thinking if there is a way I could run tests for only the files that are changed? Will this work or create problems. Also How should I go about implementing this, I couldn't find any CLI command for this specific purpose.

I checked around vitest docs, and found out this

``` 
maxConcurrency

    Type: number

    Default: 5

    CLI: --max-concurrency=10, --maxConcurrency=10

A number of tests that are allowed to run at the same time marked with test.concurrent.

Test above this limit will be queued to run when available slot appears.
```

Will this work? This would still run for the entire codebase?

Which approach should I go with?

**Comment by sheremet-va** [comment]
From your description, I think what you are looking for is `vitest related` command, or `vitest --changed=$commit`: https://vitest.dev/guide/cli.html#vitest-related

**Reply by karthxk07** [reply]
Thank you. I will try this one. 

But is it a good practice, because then it might not run tests for other affected files that are not really changed.
